### PR TITLE
106 edges refactor

### LINKED ISSUE #106: Refactor edge-related code
This issue is a follow-up to the merge of Aydar's edge-related code into `develop` (see https://github.com/Chaste/Chaste/pull/8).

The task is to refactor the new code so that it adheres to Chaste code conventions, in particular regarding variable and method names and documentation.

### PR DESCRIPTION
**Pull request by AlexFletcher**
This closes #106 by tidying up much of the edge-related code, including:

- adding documentation to classes, member variables and methods;
- changing method, parameter, and member variable names so they adhere to Chaste code conventions;
- minor edits to make the code more consistent with trunk, e.g. change `unsigned int` to `unsigned` throughout.-

### CODE REVIEW

File: cell_based/src/cell/Cell.hpp [hunk 1]
```diff
@@ -151,8 +151,9 @@ class Cell : private boost::noncopyable, public boost::enable_shared_from_this<C
     /** Whether the cell is being tracked specially. */
     bool mIsLogged;
 
-    /**Whether the cell has SRN associated to it*/
+    /** Whether the cell has SRN associated to it. */
```
Comment (mirams) [inline cell_based/src/cell/Cell.hpp]:
```suggestion
    /** Whether the cell has a sub-cellular reaction network (SRN) system of ODEs associated with it. */
```

File: cell_based/src/cell/Cell.hpp [hunk 2]
```diff
@@ -430,7 +431,7 @@ class Cell : private boost::noncopyable, public boost::enable_shared_from_this<C
     unsigned GetCellId() const;
 
     /**
-     * @return True if cell has SRN model associated to it
+     * @return whether the cell has an SRN model associated to it.
```
Comment (mirams) [inline cell_based/src/cell/Cell.hpp]:
```suggestion
     * @return Whether the cell has a sub-cellular reaction network (SRN) system of ODEs associated with it
```

File: cell_based/src/cell/srn/CellSrnModel.hpp [hunk 3]
```diff
@@ -109,75 +112,82 @@ class CellSrnModel : public AbstractSrnModel
...
+    virtual void Initialise();
 
     /**
-     * Calls SRN model specific behaviour at the time of cell division
+     * Calls SRN model specific behaviour at the time of cell division.
      * All constituent SRNs models (edge and/or interior, if there are any) call their
-     * implementation of this method
+     * implementation of this method.
      */
-    virtual void ResetForDivision() override;
```
Comment (mirams) [inline cell_based/src/cell/srn/CellSrnModel.hpp:121]: Do we have a policy on these new `override` things? From what I have read a good idea to include, but more for human and preventing human error than for the compiler?

File: cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp [hunk 4]
```diff
@@ -172,40 +183,42 @@ class DeltaNotchEdgeSrnModel : public AbstractOdeSrnModel
...
      */
-    virtual void AddMergedEdgeSrn(AbstractSrnModel* p_merged_edge_srn) override;
+    virtual void AddMergedEdgeSrn(AbstractSrnModel* pMergedEdgeSrn);
 
     /**
      * By default, Edge concentrations are split according to relative lengths, when an edge is split.
      * Override the method declared in AbstractSrnModel class
-     * @param relative_position
+     * 
+     * @param relativePosition relative position to use when splitting Edge concentrations
```
Comment (mirams) [inline cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp]: Can we explain what this `relativePosition` means a bit more? This is a bit of a tautology comment!

**Review by mirams — COMMENTED** [review]
A couple of comments above!

File: cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp (hunk 4 above)
Comment (BJackal) [reply cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp]: This makes sense as it is already explained a little bit more in the AbstractSrnModel which this class inherits from. Abstract has it define this way in the comment: 

"Scales SRN quantities due to edge split. Amount of scaling may depend on the resulting proportions, determined by relative position of the node that splits the edge."

So something along the lines of "Relative position of the new node which splits the two edges" or something along those lines could be added here. Do you think that is sufficient ?

File: cell_based/src/cell/srn/CellSrnModel.hpp (hunk 3 above)
Comment (AlexFletcher) [reply cell_based/src/cell/srn/CellSrnModel.hpp:121]: One for @fcooper8472 to comment on? I removed them here for consistency with the rest of the codebase, but if we want to include them everywhere that's fine (and probably should be part of another issue related to more extensive use of modern C++, like `auto`).

File: cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp (hunk 4 above)
Comment (mirams) [reply cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp]: Yeah, it wants to say relative position of what relative to what.

Comment (AlexFletcher) [reply cell_based/src/cell/srn/DeltaNotchEdgeSrnModel.hpp]: Changes required by @mirams were made by @BJackal in commits abd66f0, f51bda9, and 88730a2.

File: cell_based/src/cell/srn/CellSrnModel.hpp (hunk 3 above)
Comment (AlexFletcher) [reply cell_based/src/cell/srn/CellSrnModel.hpp:121]: @fcooper8472 says we should have override in all of the appropriate places, but we'll do this as part of another issue looking at modernising the C++, use of keywords etc.

**Comment by AlexFletcher** [comment]
Hi @mirams please could you re-review now?

File: cell_based/src/cell/properties/CellEdgeData.hpp [hunk 5]
```diff
@@ -89,31 +88,31 @@ class CellEdgeData : public AbstractCellProperty
...
      * @return An array of cell edge data
      */
     std::vector<double> GetItem(const std::string& rVariableName) const;
 
     /**
      * Retrieves the data of rVariableName at index
      *
-     * @param rVariableName
-     * @param index
+     * @param rVariableName the name of the data to be set.
```
Comment (mirams) [inline cell_based/src/cell/properties/CellEdgeData.hpp]:
```suggestion
     * @param rVariableName the name of the data to get.
```

**Review by mirams — APPROVED** [review]

**Pull request merged**